src/com/boxboat/jenkins/pipeline/deploy/BoxDeploy.groovy

```diff
@@ -223,8 +238,20 @@ class BoxDeploy extends BoxBase<DeployConfig> implements Serializable {
         }
     }
 
-    def withCredentials(closure) {
-        deployTarget.withCredentials(closure)
+    DeployType getDeployType(){
+        return deployType
+    }
+
+    List<DeployTarget> getDeployTargets() {
+        return config.deployTargets
+    }
+
+    def withCredentials(String targetKey="", closure) {
```

Comment:
I do not think this is useful.  What would be more useful is iterating through `deployTargetRefs`.  Probably best to just leave the `withCredentials` method alone.

Reply:
So how do we do that in an extensible generic way? For targeted first customer, I am going to need to pass `values-<deployTargetRef.name>.yaml` to `helm` - so the closure will need to be different for each `deployTargetRef`.